packages/jest-mock/src/index.ts

```diff
@@ -183,16 +185,6 @@ type MockFunctionConfig = {
   specificMockImpls: Array<Function>;
 };
 
-// see https://github.com/Microsoft/TypeScript/issues/25215
```

Comment:
The comment here was documenting the `& string` workaround. Seems like this was TS v2 issue, it does not reproduce in v3 and up. Safe to remove. Of course, I added type tests to be sure.